rust-install/src/errors.rs

```diff
@@ -41,6 +43,10 @@ pub enum Error {
 	ExtensionNotFound(rust_manifest::Component),
 	InvalidChangeSet,
 	NoGPG,
+    InstallerVersion(String),
+    InstalledMetadataVersion(String),
+    WalkDirForPermissions(walkdir::Error),
+    SetPermissions(io::Error),
```

Comment:
I generally prefer to use error names to describe the problem, rather than the action during which the problem occurred. (eg. BadInstallerVersion) although I concede that this can result in lengthier error names. (It makes it a lot easier for the caller to handle specific error-cases, and the error messages to the user can be a lot friendlier)